<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<meta http-equiv="x-ua-compatible" content="ie=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

<meta property="og:title" content="How to demultiplex fastq files with a dedicated, separate barcode file" />
<meta property="og:description" content="Oh, Barcodes, How do I obtain thee? Let me count the ways:
1) Internal to one of the reads
2) Printed in the header:
3) Encoded in the header
4) In a separate file
There are many tools out there to deal with multiplexed read data.
fastx_barcode_splitter.pl(from FASTX_Toolkit)
fastq-multx (from ea-utils)
split_libraries_fastq.py (from QIIME)
As far as I know, none of the above adequately deal with case (4) above. The QIIME tools does, but appears to turn the fastq into a fasta file in the process (?" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://biocozy.net/posts/2012-10-16-how-to-demultiplex-fastq-files-with-a-dedicated-separate-barcode-file/" />



<meta property="article:published_time" content="2012-10-16T12:12:00-07:00"/>

<meta property="article:modified_time" content="2012-10-16T12:12:00-07:00"/>











<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="How to demultiplex fastq files with a dedicated, separate barcode file"/>
<meta name="twitter:description" content="Oh, Barcodes, How do I obtain thee? Let me count the ways:
1) Internal to one of the reads
2) Printed in the header:
3) Encoded in the header
4) In a separate file
There are many tools out there to deal with multiplexed read data.
fastx_barcode_splitter.pl(from FASTX_Toolkit)
fastq-multx (from ea-utils)
split_libraries_fastq.py (from QIIME)
As far as I know, none of the above adequately deal with case (4) above. The QIIME tools does, but appears to turn the fastq into a fasta file in the process (?"/>



    <link rel="canonical" href="https://biocozy.net/posts/2012-10-16-how-to-demultiplex-fastq-files-with-a-dedicated-separate-barcode-file/">

    <title>
      
        How to demultiplex fastq files with a dedicated, separate barcode file | Biocozy.net
      
    </title>

    
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/css/bootstrap.min.css" integrity="sha384-WskhaSGFgHYWDcbwN70/dfYBj47jz9qbsMId/iRN3ewGhXQFZCSftd1LZCfmhktB" crossorigin="anonymous">

    <link href="https://biocozy.net/css/style.css" rel="stylesheet">

    

    

    
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-120822435-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-120822435-1');
</script>

  </head>
  <body>
    
      <header class="blog-header">
    <nav class="navbar navbar-expand-md navbar-light bg-light">
        <a class="navbar-brand" href="/">
            Biocozy.net
        </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false"
            aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse justify-content-between" id="navbarNav">
            <ul class="navbar-nav">
                
                
                <li class="nav-item ">
                    <a class="nav-link" href="/">Home</a>
                </li>
                
                <li class="nav-item ">
                    <a class="nav-link" href="/about/">About</a>
                </li>
                
                <li class="nav-item ">
                    <a class="nav-link" href="/contact/">Contact</a>
                </li>
                
            </ul>
            
        </div>
    </nav>
</header>
    

    
    <div class="container">
      <div class="row">
        <div class="col-12 col-lg-8 blog-main">

          

<header>
    <h2 class="blog-post-title">
        <a class="text-dark" href="/posts/2012-10-16-how-to-demultiplex-fastq-files-with-a-dedicated-separate-barcode-file/">How to demultiplex fastq files with a dedicated, separate barcode file</a>
    </h2>
    


<div class="blog-post-date text-secondary">
    
        Oct 16, 2012
    
    
        by <span rel="author">Owen</span>
    
</div>

    
    
    <hr>
</header>
<article class="blog-post">
    <p>Oh, Barcodes, How do I obtain thee?  Let me count the ways:<br />
1) Internal to one of the reads<br />
2) Printed in the header:<br />
3) Encoded in the header<br />
4) In a separate file</p>

<p>There are many tools out there to deal with multiplexed read data.<br />
<a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_barcode_splitter_usage">fastx_barcode_splitter.pl</a>(from
FASTX_Toolkit)<br />
<a href="http://code.google.com/p/ea-utils/wiki/FastqMultx">fastq-multx</a> (from
ea-utils)<br />
<a href="http://qiime.org/scripts/split_libraries_fastq.html">split_libraries_fastq.py</a>
(from QIIME)</p>

<p>As far as I know, none of the above adequately deal with case (4) above.
 The QIIME tools does, but appears to turn the fastq into a fasta file
in the process (?).</p>

<p>Here is a one-liner homebrew solution to this situation:  just tack the
index reads onto the front of read1, and then use an off-the-shelf tool
(I like the FASTX_Toolkit one above) to demultiplex.  Like this:</p>

<p>paste -d &ldquo; &lt;(echo; sed -n &lsquo;1,${n;p;}&lsquo; <span
style="background-color: yellow;">R2.fq</span> | sed G) <span
style="background-color: lime;">R1.fq</span> | sed &lsquo;/^$/d&rsquo; |
fastx_barcode_splitter.pl &ndash;bol &ndash;bcfile <span
style="background-color: #cfe2f3;">barcodes.txt</span> &ndash;prefix
debarcoded</p>

<p>I am assuming that <span
style="background-color: yellow;">R2.fq</span> is the barcode fastq
file, and <span style="background-color: lime;">R1.fq</span> is the is
the Read 1 file.  The commands above take the 2nd and 4th lines from the
barcode fastq file, spaces them out correctly, pastes them together
(with no delimiter - note: &ldquo; is a double single-quote, not a single
double-quote), and then uses another paste command to pre-pend THAT
result to the R1 file.  Also there is a final sed command to remove the
trailing blank lines.  It ain&rsquo;t pretty, but it works.</p>

<p>The fancy &lt;(  ) constructs is a bash thing called &ldquo;<a href="http://tldp.org/LDP/abs/html/process-sub.html">process
substitution</a>&rdquo;.  If
you&rsquo;re not using bash, it won&rsquo;t work.</p>

<p>Hope this helps someone.</p>

<p>(edit:  I simplified the initial sed command on 2/1/2013, so let me know
if it somehow broke, but it should still work fine.)</p>


    

    


</article>



        </div>

        <aside class="col-12 col-lg-3 ml-auto blog-sidebar">
    
        


    
    
        <section>
    
        
    
        
    
        
    
        
        <h4>Tags</h4>
        <p>
            
            <a class="badge badge-primary" href="/tags/algorithms">algorithms</a>
            
            <a class="badge badge-primary" href="/tags/annotation">annotation</a>
            
            <a class="badge badge-primary" href="/tags/bioconductor">bioconductor</a>
            
            <a class="badge badge-primary" href="/tags/bioinformatics">bioinformatics</a>
            
            <a class="badge badge-primary" href="/tags/cart">cart</a>
            
            <a class="badge badge-primary" href="/tags/computers">computers</a>
            
            <a class="badge badge-primary" href="/tags/data-mining">data-mining</a>
            
            <a class="badge badge-primary" href="/tags/genomics">genomics</a>
            
            <a class="badge badge-primary" href="/tags/geo">geo</a>
            
            <a class="badge badge-primary" href="/tags/installation">installation</a>
            
            <a class="badge badge-primary" href="/tags/linux">linux</a>
            
            <a class="badge badge-primary" href="/tags/mac-os-x">mac-os-x</a>
            
            <a class="badge badge-primary" href="/tags/machine-learning">machine-learning</a>
            
            <a class="badge badge-primary" href="/tags/ngs">ngs</a>
            
            <a class="badge badge-primary" href="/tags/parallel-computing">parallel-computing</a>
            
            <a class="badge badge-primary" href="/tags/perl">perl</a>
            
            <a class="badge badge-primary" href="/tags/r">r</a>
            
            <a class="badge badge-primary" href="/tags/r-project">r-project</a>
            
            <a class="badge badge-primary" href="/tags/rnaseq">rnaseq</a>
            
            <a class="badge badge-primary" href="/tags/shell">shell</a>
            
            <a class="badge badge-primary" href="/tags/ubuntu">ubuntu</a>
            
            <a class="badge badge-primary" href="/tags/work">work</a>
            
            <a class="badge badge-primary" href="/tags/workaround">workaround</a>
            
        </p>
        
    
</section>
    
</aside>

      </div>
    </div>
    

    
      






<footer class="blog-footer w-100">
    <nav class="navbar navbar-light bg-light">
        <p class="w-100 text-center">Copyright Biocozy.net</p>
        <p class="w-100 text-center"><a href="#">Back to top</a></p>
    </nav>
</footer>

    

    
    
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js" integrity="sha384-smHYKdLADwkXOn1EmN1qk/HfnUcbVRZyYmZ4qpPea6sjB/pTJ0euyQp0Mk8ck+5T" crossorigin="anonymous"></script>
  </body>
</html>
