<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<meta http-equiv="x-ua-compatible" content="ie=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

<meta property="og:title" content="What your father never told you about tophat" />
<meta property="og:description" content="Following up on yesterday&rsquo;s posting about the &ldquo;right&rdquo; way to run cufflinks, I am here recording my experiences running tophat with and without the &ndash;no-novel-juncs setting.
Same dataset as described in the previous posting. Still using the Illumina iGenomes Ensembl annotation files. The command I used was like this (note: the transcriptome index was pre-built during a previous run.)
tophat [&ndash;no-novel-juncs] -G genes.gtf &ndash;transcriptome-index=transcriptome -p 8 -o tophat_out file_1.fq.gz file_2.fq.gz" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://biocozy.net/posts/2013-02-01-what-your-father-never-told-you-about-tophat/" />



<meta property="article:published_time" content="2013-02-01T14:16:00-08:00"/>

<meta property="article:modified_time" content="2013-02-01T14:16:00-08:00"/>











<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="What your father never told you about tophat"/>
<meta name="twitter:description" content="Following up on yesterday&rsquo;s posting about the &ldquo;right&rdquo; way to run cufflinks, I am here recording my experiences running tophat with and without the &ndash;no-novel-juncs setting.
Same dataset as described in the previous posting. Still using the Illumina iGenomes Ensembl annotation files. The command I used was like this (note: the transcriptome index was pre-built during a previous run.)
tophat [&ndash;no-novel-juncs] -G genes.gtf &ndash;transcriptome-index=transcriptome -p 8 -o tophat_out file_1.fq.gz file_2.fq.gz"/>



    <link rel="canonical" href="https://biocozy.net/posts/2013-02-01-what-your-father-never-told-you-about-tophat/">

    <title>
      
        What your father never told you about tophat | Biocozy.net
      
    </title>

    
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/css/bootstrap.min.css" integrity="sha384-WskhaSGFgHYWDcbwN70/dfYBj47jz9qbsMId/iRN3ewGhXQFZCSftd1LZCfmhktB" crossorigin="anonymous">

    <link href="https://biocozy.net/css/style.css" rel="stylesheet">

    

    

    
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-120822435-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-120822435-1');
</script>

  </head>
  <body>
    
      <header class="blog-header">
    <nav class="navbar navbar-expand-md navbar-light bg-light">
        <a class="navbar-brand" href="/">
            Biocozy.net
        </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false"
            aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse justify-content-between" id="navbarNav">
            <ul class="navbar-nav">
                
                
                <li class="nav-item ">
                    <a class="nav-link" href="/">Home</a>
                </li>
                
                <li class="nav-item ">
                    <a class="nav-link" href="/about/">About</a>
                </li>
                
                <li class="nav-item ">
                    <a class="nav-link" href="/contact/">Contact</a>
                </li>
                
            </ul>
            
        </div>
    </nav>
</header>
    

    
    <div class="container">
      <div class="row">
        <div class="col-12 col-lg-8 blog-main">

          

<header>
    <h2 class="blog-post-title">
        <a class="text-dark" href="/posts/2013-02-01-what-your-father-never-told-you-about-tophat/">What your father never told you about tophat</a>
    </h2>
    


<div class="blog-post-date text-secondary">
    
        Feb 1, 2013
    
    
        by <span rel="author">Owen</span>
    
</div>

    
    
    <hr>
</header>
<article class="blog-post">
    <p>Following up on yesterday&rsquo;s <a href="http://biocozy.blogspot.com/2013/01/what-your-mother-never-told-you-about.html">posting about the &ldquo;right&rdquo; way to run
cufflinks</a>,
I am here recording my experiences running tophat with and without the
&ndash;no-novel-juncs setting.</p>

<p>Same dataset as described in the previous posting.  Still using the
Illumina iGenomes Ensembl annotation files. The command I used was like
this (note: the transcriptome index was pre-built during a previous
run.)</p>

<p>tophat [&ndash;no-novel-juncs] -G genes.gtf
&ndash;transcriptome-index=transcriptome -p 8 -o tophat_out file_1.fq.gz
file_2.fq.gz</p>

<p>Mean run time for the three replicates (which have very nearly the same
number of reads, about 19 million each):<br />
Default (with novel junction finding): 160 minutes<br />
With &ndash;no-novel-juncs option:  150 minutes</p>

<p>No big time savings from turning off novel junction finding.  No
difference in the mean correlation of genes.fpkm_tracking from the
resultant cufflinks run.</p>

<p>Also, I tried &ndash;no-novel-juncs with 8, 16, and 24 processors, on a
server with 32 cores (not hyperthreaded).</p>

<p>8 procs = 150 minutes<br />
16 procs = 134 minutes<br />
24 procs = 121 minutes</p>

<p>Tophat basically parallelizes the bowtie2 alignment stage, but there is
a lot of time spent reading and writing files, which is
single-threaded.</p>

<p>I have, in the past, had problems with tophat <a href="http://seqanswers.com/forums/showthread.php?t=24995">dying when running with
too many
processors</a> (24,
for instance) but the failure is not predictable:</p>

<p><span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;">[2013-02-05
08:08:09] Mapping right_kept_reads.m2g_um_seg2 to genome
segment_juncs with Bowtie2 (<sup>2</sup>&frasl;<sub>4</sub>)</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;">[2013-02-05
08:08:34] Mapping right_kept_reads.m2g_um_seg3 to genome
segment_juncs with Bowtie2 (<sup>3</sup>&frasl;<sub>4</sub>)</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;">[2013-02-05
08:09:01] Mapping right_kept_reads.m2g_um_seg4 to genome
segment_juncs with Bowtie2 (<sup>4</sup>&frasl;<sub>4</sub>)</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;">[2013-02-05
08:09:25] Joining segment hits</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;">[2013-02-05
08:11:22] Reporting output tracks</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;"> 
      [FAILED]</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;">Error
running /usr/local/tophat-2.0.7.Linux_x86_64/tophat_reports
&ndash;min-anchor 8 &ndash;splice-mismatches 0 &ndash;min-report-intron 50
&ndash;max-report-intron 500000 &ndash;min-isoform-fraction 0.15 &ndash;output-dir
Sample_04_L005/ &ndash;max-multihits 20 &ndash;max-seg-multihits 40
&ndash;segment-length 25 &ndash;segment-mismatches 2 &ndash;min-closure-exon 100
&ndash;min-closure-intron 50 &ndash;max-closure-intron 5000 &ndash;min-coverage-intron
50 &ndash;max-coverage-intron 20000 &ndash;min-segment-intron 50
&ndash;max-segment-intron 500000 &ndash;read-mismatches 2 &ndash;read-gap-length 2
&ndash;read-edit-dist 2 &ndash;read-realign-edit-dist 3 &ndash;max-insertion-length 3
&ndash;max-deletion-length 3 -z gzip -p24 &ndash;inner-dist-mean 50
&ndash;inner-dist-std-dev 20 &ndash;gtf-annotations
/data/genomes/iGenomes/Ensembl/transcriptome.gff &ndash;gtf-juncs
Sample_04_L005/tmp/transcriptome.juncs &ndash;no-closure-search
&ndash;no-coverage-search &ndash;no-microexon-search &ndash;sam-header
Sample_04_L005/tmp/genome_genome.bwt.samheader.sam
&ndash;report-discordant-pair-alignments &ndash;report-mixed-alignments
&ndash;samtools=/usr/local/samtools/samtools &ndash;bowtie2-max-penalty 6
&ndash;bowtie2-min-penalty 2 &ndash;bowtie2-penalty-for-N 1
&ndash;bowtie2-read-gap-open 5 &ndash;bowtie2-read-gap-cont 3
&ndash;bowtie2-ref-gap-open 5 &ndash;bowtie2-ref-gap-cont 3
/data/genomes/iGenomes/Ensembl/genome.fa Sample_04_L005/junctions.bed
Sample_04_L005/insertions.bed Sample_04_L005/deletions.bed
Sample_04_L005/fusions.out Sample_04_L005/tmp/accepted_hits
Sample_04_L005/tmp/left_kept_reads.m2g.bam,Sample_04_L005/tmp/left_kept_reads.m2g_um.mapped.bam,Sample_04_L005/tmp/left_kept_reads.m2g_um.candidates
Sample_04_L005/tmp/left_kept_reads.bam
Sample_04_L005/tmp/right_kept_reads.m2g.bam,Sample_04_L005/tmp/right_kept_reads.m2g_um.mapped.bam,Sample_04_L005/tmp/right_kept_reads.m2g_um.candidates
Sample_04_L005/tmp/right_kept_reads.bam</span><br />
<span
style="font-family: Courier New, Courier, monospace; font-size: xx-small;"><strong>Error:
bam_merge failed to open BAM file
Sample_04_L005/tmp/right_kept_reads.m2g_um.candidates15.bam</strong></span></p>


    

    


</article>



        </div>

        <aside class="col-12 col-lg-3 ml-auto blog-sidebar">
    
        


    
    
        <section>
    
        
    
        
    
        
    
        
        <h4>Tags</h4>
        <p>
            
            <a class="badge badge-primary" href="/tags/algorithms">algorithms</a>
            
            <a class="badge badge-primary" href="/tags/annotation">annotation</a>
            
            <a class="badge badge-primary" href="/tags/bioconductor">bioconductor</a>
            
            <a class="badge badge-primary" href="/tags/bioinformatics">bioinformatics</a>
            
            <a class="badge badge-primary" href="/tags/cart">cart</a>
            
            <a class="badge badge-primary" href="/tags/computers">computers</a>
            
            <a class="badge badge-primary" href="/tags/data-mining">data-mining</a>
            
            <a class="badge badge-primary" href="/tags/genomics">genomics</a>
            
            <a class="badge badge-primary" href="/tags/geo">geo</a>
            
            <a class="badge badge-primary" href="/tags/installation">installation</a>
            
            <a class="badge badge-primary" href="/tags/linux">linux</a>
            
            <a class="badge badge-primary" href="/tags/mac-os-x">mac-os-x</a>
            
            <a class="badge badge-primary" href="/tags/machine-learning">machine-learning</a>
            
            <a class="badge badge-primary" href="/tags/ngs">ngs</a>
            
            <a class="badge badge-primary" href="/tags/parallel-computing">parallel-computing</a>
            
            <a class="badge badge-primary" href="/tags/perl">perl</a>
            
            <a class="badge badge-primary" href="/tags/r">r</a>
            
            <a class="badge badge-primary" href="/tags/r-project">r-project</a>
            
            <a class="badge badge-primary" href="/tags/rnaseq">rnaseq</a>
            
            <a class="badge badge-primary" href="/tags/shell">shell</a>
            
            <a class="badge badge-primary" href="/tags/ubuntu">ubuntu</a>
            
            <a class="badge badge-primary" href="/tags/work">work</a>
            
            <a class="badge badge-primary" href="/tags/workaround">workaround</a>
            
        </p>
        
    
</section>
    
</aside>

      </div>
    </div>
    

    
      






<footer class="blog-footer w-100">
    <nav class="navbar navbar-light bg-light">
        <p class="w-100 text-center">Copyright Biocozy.net</p>
        <p class="w-100 text-center"><a href="#">Back to top</a></p>
    </nav>
</footer>

    

    
    
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js" integrity="sha384-smHYKdLADwkXOn1EmN1qk/HfnUcbVRZyYmZ4qpPea6sjB/pTJ0euyQp0Mk8ck+5T" crossorigin="anonymous"></script>
  </body>
</html>
